sensor/kubernetes/certrefresh/certificates/requester.go

```diff
@@ -187,37 +137,36 @@ func (r *genericRequester[ReqT, RespT]) RequestCertificates(ctx context.Context)
 		}
 	}
 
+	if r.ongoingRequest.Load() {
+		return nil, errors.New("concurrent requests are not supported.")
+	}
+	r.ongoingRequest.Store(true)
+	defer func() {
+		r.ongoingRequest.Store(false)
+	}()
+
 	requestID := uuid.NewV4().String()
-	receiveC := make(chan RespT, 1)
-	r.requests.Store(requestID, receiveC)
-	defer r.requests.Delete(requestID)
-	// Always delete this entry when leaving this scope to account for requests that are never responded, to avoid
-	// having entries in `r.requests` that are never removed.
+	r.ongoingRequestID = requestID
+	r.responseReceived.Reset()
```

Comment:
Should we maybe put this line into the `defer func` from line 144?
It is kind of the same thing, just one signals that the object is free and can accept next tasks, and the other says that there the current task is gone.